AK/SIMD.h

```diff
@@ -10,6 +10,88 @@
 
 namespace AK::SIMD {
 
+#if !defined(KERNEL) && ARCH(X86_64)
+#    define SIMD_CAN_POSSIBLY_SUPPORT_AVX2 1
+#else
+#    define SIMD_CAN_POSSIBLY_SUPPORT_AVX2 0
+#endif
+
+// FIXME: Support at least x86-64 SSE4.2, AVX512; Aarch64 SIMD
+enum class UnrollingMode {
+    NONE = 0,
+#if SIMD_CAN_POSSIBLY_SUPPORT_AVX2
+    AVX2 = 1 << 0,
+#else
+    AVX2 = NONE,
+#endif
+};
+using enum UnrollingMode;
+
+inline u32 get_supported_unrolling_modes()
+{
+#ifdef KERNEL
+    // It looks as if we do not want to use SIMD registers in the kernel code.
+    return static_cast<u32>(NONE);
+#elif ARCH(X86_64)
+    // I don't want to include Optional.h just for this one variable.
+    static bool has_cached = false;
+    static u32 cached = static_cast<u32>(NONE);
+    if (has_cached)
+        return cached;
+
+    __builtin_cpu_init();
+    if (__builtin_cpu_supports("avx2"))
+        cached |= static_cast<u32>(AVX2);
+
+    has_cached = true;
+    return cached;
```

Comment:
```suggestion
    static u32 s_cached = [] {
        if (__builtin_cpu_supports("avx2"))
            return static_cast<u32>(AVX2);
        return static_cast<u32>(NONE);
    } ();
    return s_cached;
```

This also suffices if you don't mind the load from `s_cached` being thread-safe/atomic. ([Godbolt](https://gcc.godbolt.org/z/K8qqYszMa))

Or the caching might not even be necessary. `__builtin_cpu_init()` is called as a global constructor anyway, so unless you're calling SIMD code before `main()`, calling the function here is redundant. And the check compiles down to very simple code:
```asm
main:
        mov     eax, DWORD PTR __cpu_model[rip+12]
        shr     eax, 10
        and     eax, 1
        ret
```

([Godbolt](https://gcc.godbolt.org/z/87Tax1exe))

Reply:
Oh, this is much simpler :) I somehow skipped the note about global constructor while reading the documentation.